### ISSUE TITLE
error during updateTeamTimers due to missing spacers between tab groups

### ISSUE THREAD

**Issue by caver456** [ncssar/radiolog#689]
this error repeats every second, after a team is created that would be in a different tab group, generating a spacer:
```
Traceback (most recent call last):
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 3814, in updateTeamTimers
    self.ui.tabWidget.tabBar().tabButton(i,QTabBar.LeftSide).setStyleSheet(statusStyleDict[status])
AttributeError: 'NoneType' object has no attribute 'setStyleSheet'
```

Either the index variable i doesn't account for spacers, or, the spacer is not getting generated correctly.

**Comment by caver456** [comment]
This error was discovered during work on #683, which is being done in a separate branch.  A few findings:
- the error exists in the master branch (3.9.0dev) but not in released 3.9.0 - so, it was introduced during work after 3.9.0, but, not during work for #683 
- the error happens if 'SAR 1' is added first and then 'Team 1' is added second, but, it does not happen if those teams are added in reverse order (Team 1 first).

After 'SAR 1' added:
![image](https://github.com/ncssar/radiolog/assets/18752102/0571a13b-bf55-4b4d-b632-7d8befb17ae9)

Then, after 'Team 1' added - the gap is missing and the error begins:
![image](https://github.com/ncssar/radiolog/assets/18752102/81a0ec22-c81b-4038-a0ab-959ce09b211e)


In the opposite sequence, after 'Team 1' added first then 'SAR 1' added second - which doesn't cause the error:
![image](https://github.com/ncssar/radiolog/assets/18752102/9b6ff9c5-8976-4f73-ba6f-34d43c767f3f)

Should probably check for related team tab grouping issues - more than two groups, variation on whether the new tab group would be to the left or the right of the existing tab group, etc.

**Comment by caver456** [comment]
After tkdiffing and testing, the error is due to this section of code:

```
		self.rebuildGroupedTabDict()
		rprint("extTeamNameList after sort:"+str(self.extTeamNameList))
		if not self.loadFlag:
			# 670 - for the first team of a new session, and when unhiding the only tab
			#  (due to display issues noted in https://github.com/ncssar/radiolog/issues/670#issuecomment-1712814526)
			#  follow the old behavior and rebuild the entire tab bar;
			#  for subsequent teams, only add the tab for the new team 
			if self.ui.tabWidget.tabBar().count()<2:
				self.rebuildTabs()
			else:
				# display issues when unhiding the rightmost tab
				#  https://github.com/ncssar/radiolog/issues/670#issuecomment-1712814526
				# so, if this would be the rightmost tab, activate a different tab first
				self.ui.tabWidget.tabBar().setCurrentIndex(0)
				self.addTab(extTeamName)
```

versus the previous version that didn't cause the error but led to slowness (see #670):

```
		self.rebuildGroupedTabDict()
		rprint("extTeamNameList after sort:"+str(self.extTeamNameList))
		if not self.loadFlag:
			self.rebuildTabs()
```
